eosio.system/src/voting.cpp

```diff
@@ -66,8 +82,15 @@ namespace eosiosystem {
       const auto& prod = _producers.get( producer, "producer not found" );
 
       _producers.modify( prod, 0, [&]( producer_info& info ){
-            info.deactivate();
+         info.deactivate();
       });
+
+      auto prod2 = _producers2.find( producer );
```

Comment:
There is no need to add the second producer object while unregistering the producer. It will be added correctly later on when registering.